client/layout/sidebar/heading.jsx

```diff
@@ -13,6 +13,12 @@ const SidebarHeading = ( { children, onClick, ...props } ) => {
 		};
 	}
 
+	const linkAttrs = { ...props };
+
+	// These are not valid HTML attribute for a <a> tag
+	delete linkAttrs.navigationLabel;
+	delete linkAttrs.url;
```

Comment:
It's nice that you cloned the object before deleting to avoid mutating the props. 👍